### PR TITLE
add support for piano analytics

### PR DESCRIPTION
**Pull request by nathangoulding**
Tracked in issue: #1769

### CODE REVIEW

**Comment by cramforce** [comment]
Moved to #2075

**Comment by avimehta** [comment]
@cramforce was this closed by mistake? I can't find the correct PR to review.

**Comment by rudygalfi** [comment]
Re-opening as I believe this was closed by mistake.

File: examples/analytics.amp.html [hunk 1]
```diff
@@ -186,6 +186,27 @@
...
+  "triggers": {
+    "defaultPageview": {
+      "on": "visible",
+      "request": "pageview"
+    }
+  }
+}
+</script>
+</amp-analytics>
+<!-- End Piano example -->
```
Comment (avimehta) [inline examples/analytics.amp.html]: There doesn't seem to be a start tag for this. Want to add 

```
<!-- Start Piano example -->
```
before the example?

File: extensions/amp-analytics/0.1/vendors.js [hunk 2]
```diff
@@ -253,6 +253,30 @@ export const ANALYTICS_CONFIG = {
...
+  'piano': {
+    'requests': {
+      'host': 'https://api-v3.tinypass.com',
+      'basePrefix': '/api/v3',
+      'baseSuffix': '&pageview_id=${pageViewId}&rand=${random}',
+      'pageview': '${host}${basePrefix}/page/track?url=${canonicalUrl}&' +
+      'referer=${documentReferrer}&content_created=${contentCreated}&' +
+      'content_author=${contentAuthor}&content_section=${contentSection}&' +
+      'timezone_offset=${timezone}&tags=${tags}&amp_url=${ampdocUrl}&' +
+      'screen=${screenWidth}x${screenHeight}'
```
Comment (avimehta) [inline extensions/amp-analytics/0.1/vendors.js]: add a trailing comma here. We are moving towards making the trailing comma required.

File: extensions/amp-analytics/0.1/vendors.js [hunk 3]
```diff
@@ -253,6 +253,30 @@ export const ANALYTICS_CONFIG = {
...
+    },
+    'triggers': {
```
Comment (avimehta) [inline extensions/amp-analytics/0.1/vendors.js]: You can remove the triggers section. Publishers will specify that in their configuration as shown in the example.

**Comment by nathangoulding** [comment]
PTAL

File: extensions/amp-analytics/0.1/vendors.js [hunk 4]
```diff
@@ -253,6 +253,24 @@ export const ANALYTICS_CONFIG = {
     }
   },
 
+  'piano': {
+    'requests': {
+      'host': 'https://api-v3.tinypass.com',
+      'basePrefix': '/api/v3',
+      'baseSuffix': '&pageview_id=${pageViewId}&rand=${random}',
```
Comment (rudygalfi) [inline extensions/amp-analytics/0.1/vendors.js]: This looks unused. Did you mean to have it on the `pageview` request below?

File: extensions/amp-analytics/0.1/vendors.js [hunk 5]
```diff
@@ -253,6 +253,24 @@ export const ANALYTICS_CONFIG = {
...
+      'pageview': '${host}${basePrefix}/page/track?url=${canonicalUrl}&' +
+      'referer=${documentReferrer}&content_created=${contentCreated}&' +
```
Comment (rudygalfi) [inline extensions/amp-analytics/0.1/vendors.js]: Please indent all of the lines under `'pageview'` definition

**Comment by rudygalfi** [comment]
A couple more small things.

@avimehta PTAL as well

File: extensions/amp-analytics/0.1/vendors.js [hunk 6]
```diff
@@ -253,6 +253,24 @@ export const ANALYTICS_CONFIG = {
...
+      'content_author=${contentAuthor}&content_section=${contentSection}&' +
+      'timezone_offset=${timezone}&tags=${tags}&amp_url=${ampdocUrl}&' +
+      'screen=${screenWidth}x${screenHeight}',
+    },
+    'transport': {
```
Comment (avimehta) [inline extensions/amp-analytics/0.1/vendors.js]: The defaults will take care of this if all the values are true. Feel free to remove this section.

**Comment by avimehta** [comment]
lgtm.

File: extensions/amp-analytics/0.1/vendors.js (hunk 6 above)
Comment (nathangoulding) [inline extensions/amp-analytics/0.1/vendors.js]: OK will do. As per the previous code review, after I remove the `transport` block, should I keep the trailing comma at the end of the `requests`?

File: extensions/amp-analytics/0.1/vendors.js (hunk 4 above)
Comment (nathangoulding) [inline extensions/amp-analytics/0.1/vendors.js]: Ah yes, thanks for catching this.

**Comment by nathangoulding** [comment]
I'll rebase again and open another PR.

**Pull request closed without merging**